src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -316,6 +320,33 @@ public void transferItemHold(UUID transactionId, String itemBarcode) {
     eventPublisher.publishEvent(MoveRequestEvent.of(transaction, item));
   }
 
+  @Override
+  public void finalcheckinItemHold(UUID transactionId, UUID servicePointId) {
+    var transaction = fetchTransactionById(transactionId);
+
+    verifyState(transaction, ITEM_RECEIVED, RECEIVE_UNANNOUNCED);
+
+    var loan = loanService.getById(transaction.getHold().getFolioLoanId());
+
+    if (loan.getAction().equalsIgnoreCase("checkedin") && loan.getStatus().getName().equals("closed")) {
+      var hold = transaction.getHold();
+      hold.setPatronId(null);
+      hold.setPatronName(null);
+      hold.setDueDateTime(null);
+      transaction.setHold(hold);
+      transaction.setState(FINAL_CHECKIN);
+      transactionRepository.save(transaction);
+
+      notifier.reportFinalCheckIn(transaction);
```

Comment:
why don't you just reuse `updateTransactionOnLoanClosure` method here

Reply:
refactored